app/src/main/java/com/infomaniak/mail/ui/MainViewModel.kt

```diff
@@ -218,9 +215,7 @@ class MainViewModel @Inject constructor(
     private fun switchToValidMailbox() = viewModelScope.launch(ioCoroutineContext) {
         MailboxController.getFirstValidMailbox(AccountUtils.currentUserId)?.let {
             AccountUtils.switchToMailbox(it.mailboxId)
-        } ?: run {
-            AccountUtils.removeUser(context, AccountUtils.currentUser!!)
-        }
+        } ?: context.startActivity(Intent(context, NoValidMailboxesActivity::class.java))
```

Comment:
Ask @sirambd what is better between startActivity and a direction and why

Reply:
Already answered